pyvista/plotting/plotting.py

```diff
@@ -4585,6 +4595,65 @@ def export_obj(self, filename):
         exporter.SetRenderWindow(self.ren_win)
         return exporter.Write()
 
+    @property
+    def _datasets(self):
+        """Return a list of all datasets associated with this plotter."""
+        datasets = []
+        for renderer in self.renderers:
+            for actor in renderer.actors.values():
+                mapper = actor.GetMapper()
+
+                # ignore any mappers whose inputs are not datasets
+                if hasattr(mapper, 'GetInputAsDataSet'):
+                    datasets.append(mapper.GetInputAsDataSet())
+
+        return datasets
+
+    def _remove_added_scalars(self):
+        """Remove any scalars added to this plotter's datasets by this plotter.
+
+        Additional point or cell data may be added to be able to correctly plot
+        scalars. This usually occurs in one of the following cases:
+
+        * ``<scalar-name>-real`` for complex data
+        * ``<scalar-name>-normed`` for normalized multi-component arrays.
+        * ``scalars-<index>`` when setting ``component=<index>`` in
+          ``add_mesh`` when plotting multi-component arrays.
```

Comment:
Just to follow the previous general cases
```suggestion
        * ``<scalars-name>-<index>`` when setting ``component=<index>`` in
          ``add_mesh`` when plotting multi-component arrays.
```